apps/web/utils/logic/showAnswer.ts

```diff
@@ -0,0 +1,77 @@
+// repo
+import { PossibleAnswer } from "@repo/data/useWalkthroughData";
+// local
+import { AnswerTypes, AnswerToCheckValueFn } from "../../stores/AnswerStore";
+import { isArray } from "../typeChecking";
+
+export const getPossibleAnswers = (
+  possibleAnswers: PossibleAnswer[],
+  getAnswerToCheckValue: AnswerToCheckValueFn,
+) => {
+  return possibleAnswers.filter((possibleAnswer) => {
+    if (!possibleAnswer.showAnswerIf) return true;
```

Comment:
Could we do something like this? 
`if (possibleAnswer.showAnswerIf == null || possibleAnswer.showAnswerIf === true) return true;
`

Reply:
Not quite because of a quirk with TypeScript and JSON. the `showAnswerIf` property can be `boolean | ShowAnswerIf[] | undefined`. But really, it can never be false. I just can't set it to `true | ShowAnswerIf[] | undefined` because of the type difference issues between TypeScript and JSON. I will update this to absorb the check below it though. LMK what you think.